code/script.js

```diff
@@ -247,97 +300,186 @@ const generateBoard = () => {
+  setSecret();
+  board.style.display = "flex";
+  winOrLose.style.display = "none";
+
 }
 
 // setting the currentQuestion object when you select something in the dropdown
 const selectQuestion = () => {
   const category = questions.options[questions.selectedIndex].parentNode.label
-  // This variable stores what option group (category) the question belongs to.
-  // We also need a variable that stores the actual value of the question we've selected.
+  const optionValue = questions.options[questions.selectedIndex].value
 
   if (category === 'hair color') {
     currentQuestion = {
       attribute: 'hairColor',
-      // value: ,
-      // 👆 add the value from the input here
-      category: category,
+      value: optionValue,
+      category: category
     }
   } else if (category === 'eye color') {
-    // Set this up your self
+      currentQuestion = {
+        attribute: 'eyeColor',
+        value: optionValue,
+        category: category
+      }
+  } else if (category === 'facial hair') {
+    currentQuestion = {
+      attribute: 'facialHair',
+      value: true,
+      category: category
+    }
   } else if (category === 'accessories') {
     currentQuestion = {
-      //attribute: ,
-      // 👆 this is the property of the booleans such as smoke, glasses and hat. add the value from the input here
-      value: true, // we're asking if this person wears a hat for exaple, so always true in the question.
-      category: category,
+      attribute: optionValue,
+      value: true,
+      category: category
+    }
+  } else if (category === 'head covering') {
+    currentQuestion = {
+      attribute: 'headCover',
+      value: true,
+      category: category
+    }
+  } else if (category === 'clothing') {
+    currentQuestion = {
+      attribute: 'clothing',
+      value: optionValue,
+      category: category
     }
   } else if (category === 'other') {
-    // Set this up your self (should be same structure as above)
+    currentQuestion = {
+      attribute: 'smoker',
+      value: true,
+      category: category
+    }
   }
+  return currentQuestion
 }
 
 // This function should be invoked when you click on 'Find Out'.
 const checkQuestion = () => {
-  // Compare the currentQuestion with the secret person.
-  // See if we should keep or remove people based on that
-  // Then invoke filterCharacters
+  selectQuestion()
+  if (currentQuestion["value"] === secret[currentQuestion["attribute"]]) {
+    keep = true;
+  } else {
+    keep = false;
+  }
+
+
+  filterCharacters(keep)
+  count += 1;
+  console.log(count)
+  countQuestion.innerHTML = `${count}`
 }
 
-// It'll filter the characters array and redraw the game board.
+// This will filter the characters array and redraw the game board.
 const filterCharacters = (keep) => {
-  // Show the correct alert message for different categories
-  if (group === 'accessories') {
-    if (keep) {
+  if (currentQuestion.category === 'accessories') {
+    if (keep === true) {
```

Comment:
It can be skipped. To check boolean variable in a if condition there is no need to put `=== true`. The suggestion and the original code are identical and behave the same in this specific case. 

```suggestion
    if (keep) {
```